Memory leak action is failing

Since at least August 2022 the memory leak task is failing with:

```js
  1) # Cell memory leaks
       Adding a cell:
     TypeError: Cannot read properties of undefined (reading 'detected')
      at file:///home/runner/work/_actions/jupyterlab/benchmarks/v1.2.2/memory-leaks/tests/utils.mjs:29:45
      at Array.map (<anonymous>)
      at expectNoLeaks (file:///home/runner/work/_actions/jupyterlab/benchmarks/v1.2.2/memory-leaks/tests/utils.mjs:29:13)
      at testScenario (file:///home/runner/work/_actions/jupyterlab/benchmarks/v1.2.2/memory-leaks/tests/utils.mjs:226:3)
      at async Context.<anonymous> (file:///home/runner/work/_actions/jupyterlab/benchmarks/v1.2.2/memory-leaks/tests/cell.test.mjs:7:5)

  2) # Notebook memory leaks
       Opening a notebook:

      AssertionError: expected true to equal false
      + expected - actual

      -true
      +false
      
      at expectNoLeaks (file:///home/runner/work/_actions/jupyterlab/benchmarks/v1.2.2/memory-leaks/tests/utils.mjs:30:8)
      at testScenario (file:///home/runner/work/_actions/jupyterlab/benchmarks/v1.2.2/memory-leaks/tests/utils.mjs:226:3)
      at async Context.<anonymous> (file:///home/runner/work/_actions/jupyterlab/benchmarks/v1.2.2/memory-leaks/tests/notebook.test.mjs:6:5)
```


One of them seems to reference:

https://github.com/jupyterlab/benchmarks/blob/56d7de5ef9beb2624f3f747ccc138f64e504e668/memory-leaks/tests/utils.mjs#L29

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -12,6 +12,11 @@ jobs:
 
     runs-on: ubuntu-22.04
 
+    strategy:
+      matrix:
+        # FIXME Workaround for https://github.com/iterative/setup-cml/issues/85
+        node-version: ['16.x']
+
     env:
       BENCHMARK_NUMBER_SAMPLES: 100
 

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -246,7 +246,7 @@ jobs:
 
       - name: Execute memory leak tests
         id: memory_leaks
-        uses: jupyterlab/benchmarks/.github/actions/memory-leak@v1.2.3
+        uses: jupyterlab/benchmarks/.github/actions/memory-leak@v1.2.4
         with:
           server_url: localhost:8888
           artifacts_name: benchmark-assets

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -246,7 +246,7 @@ jobs:
 
       - name: Execute memory leak tests
         id: memory_leaks
-        uses: jupyterlab/benchmarks/.github/actions/memory-leak@v1.2.2
+        uses: jupyterlab/benchmarks/.github/actions/memory-leak@v1.2.3
         with:
           server_url: localhost:8888
           artifacts_name: benchmark-assets

--- a/packages/codeeditor/src/editor.ts
+++ b/packages/codeeditor/src/editor.ts
@@ -218,6 +218,7 @@ export namespace CodeEditor {
         return;
       }
       this._isDisposed = true;
+      this._selections.dispose();
       if (this.standaloneModel) {
         this.sharedModel.dispose();
       }

--- a/packages/codeeditor/src/jsoneditor.ts
+++ b/packages/codeeditor/src/jsoneditor.ts
@@ -150,6 +150,7 @@ export class JSONEditor extends Widget {
       return;
     }
 
+    this.source?.dispose();
     this.model.dispose();
     this.editor.dispose();
 

--- a/packages/notebook/src/notebooktools.ts
+++ b/packages/notebook/src/notebooktools.ts
@@ -757,6 +757,7 @@ export namespace NotebookTools {
         this.editor.source.changed.disconnect(this._onSourceChanged, this);
       }
       const nb = this.notebookTools.activeNotebookPanel?.content;
+      this.editor.source?.dispose();
       this.editor.source = nb?.model?.metadata
         ? new ObservableJSON({ values: nb.model.metadata as JSONObject })
         : null;
@@ -809,6 +810,7 @@ export namespace NotebookTools {
         this.editor.source.changed.disconnect(this._onSourceChanged, this);
       }
       const cell = this.notebookTools.activeCell;
+      this.editor.source?.dispose();
       this.editor.source = cell
         ? new ObservableJSON({ values: cell.model.metadata as JSONObject })
         : null;

